### ISSUE TITLE
YAML Syntax highlignting doesnt include multiline scalars anymore

### ISSUE THREAD

**Issue by MaxRink** [microsoft/vscode#99025]
- VSCode Version: 1.45.1
- OS Version: Win10x86

Steps to Reproduce:
In previous versions vscode linted yaml in  multiline scalars, now it doesnt anymore:
This happens without ANY plugins installed.

1.45.1
![image](https://user-images.githubusercontent.com/4894735/83263623-32b88380-a1bf-11ea-964a-5ce9c9e1d7d6.png)

1.39.1
![image](https://user-images.githubusercontent.com/4894735/83263568-1f0d1d00-a1bf-11ea-805b-b7fdd6f46a76.png)

It also happens with the insider build (e.g. 1.46)

Does this issue occur when all extensions are disabled?: Yes

**Comment by alexr00** [comment]
Can you please provide a copy/paste-able example?

**Comment by MaxRink** [comment]
sure
```yaml
---
apiVersion: v1
kind: ConfigMap
metadata:
  name: nginx-ingress-controller-customer-configmap
  namespace: ingress-system
data:
  values.yaml: |
    service:
      annotations:
        metallb.universe.tf/allow-shared-ip: "10.27.50.35/32"
      loadBalancerIP: "10.27.50.35"
```

**Comment by alexr00** [comment]
I can see the repro. This is very odd though, since the yaml grammar hasn't changed since 2017. It looks like this started with version 1.41.
Starting with 1.41, the scope for `service` in your example is `string.unquoted.block.yaml`.
In 1.40, the scope of `service` was `entity.name.tag.yaml`, `string.unquoted.plain.out.yaml`.

It looks like whatever grammar that GitHub is using does the same thing. 

@alexdima, before I spend more time looking into this, do you know of any changes vscode textmate that started with 1.41 that could have changed this?

**Comment by alexdima** [comment]
Yes, here is what steps I used to track them down.

* `1.40` shipped with `vscode-textmate@4.2.2` -- https://github.com/microsoft/vscode/blob/release/1.40/package.json#L55
* `1.41` shipped with `vscode-textmate@4.4.0` -- https://github.com/microsoft/vscode/blob/release/1.41/package.json#L55

If using tags to publish to `npm` (which `vscode-textmate` does), it is possible to use github to find out the changes: https://github.com/microsoft/vscode-textmate/compare/v4.2.2...v4.4.0

From looking at those changes, this change mentions yaml explicitly -- https://github.com/microsoft/vscode-textmate/issues/119

I can use the inspect tool to investigate.

**Comment by alexdima** [comment]
This appears to be correct. This is a direct consequence of the fix for https://github.com/microsoft/vscode-textmate/issues/119 . 

It is expected that the fix for https://github.com/microsoft/vscode-textmate/issues/119 changes this between VS Code versions. The new behaviour is aligned with how TextMate parses using this same grammar:

![image](https://user-images.githubusercontent.com/5047891/84022700-fc30f480-a986-11ea-88d5-cae1c7c97d6f.png)

So **TL;DR** the old version of vscode appears to look better, but it was wrong.
